users/views.py

```diff
@@ -164,4 +171,279 @@ def get(self, request, user_id, token):
+
+            user, _ = self.get_or_create_user(
+                email=email,
+                defaults={
+                    # "username": username,
+                    "first_name": first_name,
+                    "last_name": last_name,
+                    "password": "",
+                    "user_phone": "",
+                    "title": "",
+                    "role": get_default_role()
+                }
+            )
+
+            return self.generate_jwt_response(user)
+            
+        except requests.RequestException:
+            return Response(
+                {"error": "Invalid GitHub token"}, 
+                status=status.HTTP_400_BAD_REQUEST
+            )
+    
+    def get_or_create_user(self, email, defaults):
+        """
+        Helper method to get or create user with intelligent field updates.
+        
+        Args:
+            email: User's email address (primary lookup key)
+            defaults: Dictionary of default values for new user creation
+            
+        Returns:
+            tuple: (user_object, created_bool) where:
+                - user_object: The retrieved or created user instance
+                - created_bool: Boolean indicating if user was created
+        """
+        user, created = User.objects.get_or_create(
+            email=email,
+            defaults=defaults
+        )
+        
+        if created:
+            user.set_unusable_password()
+            user.save()
+        else:
+            update_fields = {}
+            if not user.first_name and defaults.get('first_name'):
+                update_fields['first_name'] = defaults['first_name']
+            if not user.last_name and defaults.get('last_name'):
+                update_fields['last_name'] = defaults['last_name']
+            
+            if update_fields:
+                User.objects.filter(pk=user.pk).update(**update_fields)
+                user.refresh_from_db()
+        
+        return user, created
+    
+    def generate_jwt_response(self, user):
+        """
+        Generate standardized JWT token response with user data.
+        
+        Args:
+            user: Authenticated user instance
+            
+        Returns:
+            Response: DRF Response object containing:
+                - refresh: JWT refresh token
+                - access: JWT access token
+                - user: Serialized user data
+        """
+        refresh = RefreshToken.for_user(user)
+        return Response({
+            "refresh": str(refresh),
+            "access": str(refresh.access_token),
+            "user": {
+                "id": user.user_id,
+                # "username": username,
+                "email": user.email,
+                "first_name": user.first_name,
+                "last_name": user.last_name,
+                "user_phone": user.user_phone,
+                "title": user.title,
+                "role": str(get_default_role())
+            }
+        })        
```

Comment:
Include a serializer for user data rather than hardcoding fields, for easier future changes.
Return the actual role field, not str(get_default_role()), to avoid confusion between the user's role and the default role.
If username is supported, uncomment and set it accordingly.